src/main/webapp/app/file-upload-submission/file-upload-submission.component.ts

```diff
@@ -0,0 +1,193 @@
+                        this.numberOfAllowedComplaints = allowedComplaints;
+                    });
+                }
+
+                if (data.submissions && data.submissions.length > 0) {
+                    this.submission = data.submissions[0] as FileUploadSubmission;
+                }
+
+                this.isActive =
+                    this.fileUploadExercise.dueDate === undefined || this.fileUploadExercise.dueDate === null || new Date() <= moment(this.fileUploadExercise.dueDate).toDate();
+            },
+            (error: HttpErrorResponse) => this.onError(error),
+        );
+    }
+
+    /**
+     * Uploads a submission file and submits File Upload Exercise
+     */
+    submit() {
+        const file = this.submissionFile;
+
+        if (!this.submission || !file) {
+            return;
+        }
+
+        this.erroredFile = null;
+        this.fileUploaderService.uploadFile(file, file['name'], { keepFileName: true }).then(
+            result => {
+                this.submission!.filePath = result.path;
+                this.submitExercise();
+            },
+            error => {
+                console.error('Error during file upload in uploadBackground()', error.message);
+                this.erroredFile = file;
+                this.fileInput.nativeElement.value = '';
+                this.submission!.filePath = null;
+                this.submissionFile = null;
+            },
+        );
+    }
+
+    /**
+     * Submits File Upload Exercise
+     */
+    submitExercise() {
+        const confirmSubmit = window.confirm(this.submissionConfirmationText);
+
+        if (confirmSubmit) {
+            this.submission!.submitted = true;
+            this.fileUploadSubmissionService.update(this.submission!, this.fileUploadExercise.id).subscribe(
+                response => {
+                    this.submission = response.body!;
+                    this.result = this.submission.result;
+
+                    if (this.isActive) {
+                        this.jhiAlertService.success('artemisApp.fileUploadExercise.submitSuccessful');
+                    } else {
+                        this.jhiAlertService.warning('artemisApp.fileUploadExercise.submitDeadlineMissed');
+                    }
+                },
+                err => {
+                    this.jhiAlertService.error('artemisApp.fileUploadExercise.error');
+                    this.submission!.submitted = false;
+                },
+            );
+        }
+    }
+
+    /**
+     * Sets file submission for exercise
+     * @param $event {object} Event object which contains the uploaded file
+     */
+    setFileSubmissionForExercise($event: any): void {
+        if ($event.target.files.length) {
+            this.erroredFile = null;
+            const fileList: FileList = $event.target.files;
+            const submissionFile = fileList[0];
+            const allowedFileExtensions = this.fileUploadExercise.filePattern.replace(/\s/g, '').split(',');
```

Comment:
This should be done during serialization/deserialization. Not super important for this PR, but a Todo for the next one.

Reply:
Moved it to the `fileUploadExerciseService`